proxy/conn_test.go

```diff
@@ -0,0 +1,189 @@
+package proxy
+
+import (
+	"bufio"
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/websocket"
+	"github.com/matrix-org/matrix-websockets-proxy/mocks/io"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+// testSendMessage tests that Conn.SendMessage causes a text message to be
+// written to the web socket
+func TestSendMessage(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	wg := &sync.WaitGroup{}
```

Comment:
It's a little strange to explicitly make a nil value; I would expect this to just be:

var wg sync.WaitGroup